Wikipedia/Code/NSString+WMFExtras.m

```diff
@@ -76,19 +76,22 @@ - (NSString *)wmf_stringBySanitizingForJavaScript {
     NSRegularExpression *regex = [NSRegularExpression wmf_charactersToEscapeForJSRegex];
     NSMutableString *mutableSelf = [self mutableCopy];
     __block NSInteger offset = 0;
-    [regex enumerateMatchesInString:self options:0 range:NSMakeRange(0, self.length) usingBlock:^(NSTextCheckingResult * _Nullable result, NSMatchingFlags flags, BOOL * _Nonnull stop) {
-        NSInteger indexForBackslash = result.range.location + offset;
-        if (indexForBackslash >= mutableSelf.length) {
-            return;
-        }
-        [mutableSelf insertString:@"\\" atIndex:indexForBackslash];
-        offset += 1;
-    }];
+    [regex enumerateMatchesInString:self
+                            options:0
+                              range:NSMakeRange(0, self.length)
+                         usingBlock:^(NSTextCheckingResult *_Nullable result, NSMatchingFlags flags, BOOL *_Nonnull stop) {
+                             NSInteger indexForBackslash = result.range.location + offset;
+                             if (indexForBackslash >= mutableSelf.length) {
+                                 return;
+                             }
+                             [mutableSelf insertString:@"\\" atIndex:indexForBackslash];
+                             offset += 1;
+                         }];
     return mutableSelf;
 }
 
 - (NSString *)wmf_stringBySanitizingForBacktickDelimitedJavascript {
-   return [self stringByReplacingOccurrencesOfString:@"([{}\\`])" withString:@"\\\\$1" options:NSRegularExpressionSearch range:NSMakeRange(0, self.length)];
+    return [self stringByReplacingOccurrencesOfString:@"([\\\\{}\\`])" withString:@"\\\\$1" options:NSRegularExpressionSearch range:NSMakeRange(0, self.length)];
```

Comment:
stupid regex question... how does adding more slashes to the bracketed character class fix this? my understanding was it previous read "match left curly brace or right curly brace or backslash" - now it's "match backslash or backslash or left curly brace or right curly brace or backslash"...

Reply:
oops never mind i see now

Reply:
i was missing the backtick